lib/event-target.js

```diff
@@ -123,7 +123,7 @@ class ErrorEvent extends Event {
   }
 
   /**
-   * @type {*}
+   * @type {Required<NonNullable<ConstructorParameters<typeof ErrorEvent>[1]>>["error"]}
```

Comment:
This can just be the direct type, but it would mean that `error` would not automatically be kept in sync with the constructor argument. Same for the `data` property in `MessageEvent`.

Reply:
I guess another option is to try to type the `error` property itself, and then make the constructor parameter `ErrorEvent["error"] | null` or something like that. (Again, same for `MessageEvent`.)

Reply:
Resolving. See https://github.com/websockets/ws/pull/2242#discussion_r1676920333